src/EmailGuardianVerifier.sol

```diff
@@ -0,0 +1,553 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {IDKIMRegistry} from "@zk-email/contracts/DKIMRegistry.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+import {IGuardianVerifier} from "./interfaces/IGuardianVerifier.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IEmailRecoveryCommandHandler} from "./interfaces/IEmailRecoveryCommandHandler.sol";
+import {EmailAuthMsg} from "@zk-email/ether-email-auth-contracts/src/EmailAuth.sol";
+import {CommandUtils} from "@zk-email/ether-email-auth-contracts/src/libraries/CommandUtils.sol";
+import {IVerifier, EmailProof} from "@zk-email/ether-email-auth-contracts/src/interfaces/IVerifier.sol";
+
+/**
+ * @title EmailGuardianVerifier
+ * @notice Provides a mechanism for guardian verification using email-based proofs.
+ * @dev The underlying IGuardianVerifier provides the interface for proof verification.
+ */
+contract EmailGuardianVerifier is IGuardianVerifier, Initializable {
```

Comment:
Proposed refactor to make this file simpler:

We have an [EmailSigner](https://github.com/zkemail/email-tx-builder/blob/main/packages/contracts/src/EmailSigner.sol) that is meant to be used as a lower level primitive for use cases such as 4337 validators, recovery modules, Safe contract signers. You could add that here which would have a lot of the logic you've written already covered.

This is likely what we would do for our account recovery v2 (build email guardians using EmailSigner).

Will publish an experimental version of our package (maybe later today) so you can access it, otherwise feel free to copy paste it.

Reply:
ok the issue with using EmailSigner is that the constructor logic will clash with the logic of `computeGuardianVerifierAddress`

Reply:
This was taking most of the base logic reference from `EmailAuth.sol` onto guardian verifier with specifics of recovery/acceptance based checks ( which we obv need to have )